backend/src/main/java/de/ffmjava/capstone/backend/user/CustomApiErrorHandler.java

```diff
@@ -3,29 +3,29 @@
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import com.mongodb.lang.Nullable;
 import lombok.Data;
 import org.springframework.validation.Errors;
 import org.springframework.validation.FieldError;
 
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
 @Data
 public class CustomApiErrorHandler {
-    @Nullable
-    private ArrayList<String> fields = new ArrayList<>();
+
     private Map<String, ArrayList<String>> fieldMessageMappings = new HashMap<>();
     private String jsonString;
     private Errors errors;
 
     public CustomApiErrorHandler(Errors errors) {
         for (FieldError singleError : errors.getFieldErrors()) {
-            if (!fields.contains(singleError.getField())) {
+            if (!fieldMessageMappings.containsKey(singleError.getField())) {
                 fieldMessageMappings.put(singleError.getField(), new ArrayList<>());
             }
             fieldMessageMappings.get(singleError.getField()).add(singleError.getDefaultMessage());
+            Collections.sort(fieldMessageMappings.get(singleError.getField()));
         }
         try {
             this.jsonString = new ObjectMapper().writeValueAsString(fieldMessageMappings);
```

Comment:
Not 100% sure but I think spring boot will automatically serialize your objects , and you don't have to do it your self.

Reply:
You are right, good eye